src/@types/events.ts

```diff
@@ -0,0 +1,35 @@
+import { Action, SerializedError } from '@reduxjs/toolkit';
+import { ApplicationState } from '../store/types';
+
+export enum Event {
```

Comment:
Should these be isolated to the middleware rather than in the global scope? Do we want to move these types to src/store/eventing/types?